app/main/controllers/shared_projects.py

```diff
@@ -0,0 +1,150 @@
+import http.client
+import os
+
+from flask import jsonify, abort, request
+
+from ..auth import requires_auth, verify_decode_jwt
+from ..models.models import Source
+from .sources import get_authorized_source
+from .items import get_authorized_item
+from .items import get_authorized_project
+
+
+def set_shared_project_routes(app):
+    """
+    Adds a new user to a shared project
+    """
+
+    @app.route('/shared_projects', methods=['POST'])
+    @requires_auth('create:shared_projects')
+    def create_share_from_ids(user_id):
+        body = request.get_json()
+        if body is None:
+            abort(400)
+
+        # Project ID to share
+        shared_proj = body.get('shared_proj', None)
+        # Email of the user to share
+        email = body.get('email', None)
+        # Role of the new user
+        role = body.get('role', None)
+
+        if shared_proj is None or email is None or role is None:
+            abort(400)
+
+        conn = http.client.HTTPSConnection("")
+
+        # Code to get MGMT token access
+        payload = "grant_type=client_credentials&client_id=%24%7Baccount.clientId%7D&client_secret=zMZtScJRq7ZSiFzvRnnMIYsIcLRDoYWmKXh56O8n4dni6kRjvPQHy6e0irQZuJao&audience=https%3A%2F%2F%24%7Baccount.namespace%7D%2Fapi%2Fv2%2F"
```

Comment:
Probably not good practice to leave a client_secret exposed here, I think the best move would be to store it in an environment variable and then draw from it (os.environ.get(). It would also be better to build this string up from individual pieces, so that we understand what each part does.